### PR TITLE
replace redundant note in deprecation warning

### PR DESCRIPTION
**Pull request by euclio**

### CODE REVIEW

File: src/test/ui/deprecation/atomic_initializers.stderr [hunk 1]
```diff
@@ -2,11 +2,7 @@ warning: use of deprecated item 'std::sync::atomic::ATOMIC_ISIZE_INIT': the `new
   --> $DIR/atomic_initializers.rs:8:27
    |
 LL | static FOO: AtomicIsize = ATOMIC_ISIZE_INIT;
-   |                           ^^^^^^^^^^^^^^^^^
+   |                           ^^^^^^^^^^^^^^^^^ help: replace the usage of the deprecated item: `AtomicIsize::new(0)`
    |
    = note: #[warn(deprecated)] on by default
-help: use of deprecated item 'std::sync::atomic::ATOMIC_ISIZE_INIT': the `new` function is now preferred
```
Comment (varkor) [inline src/test/ui/deprecation/atomic_initializers.stderr]: Why has this been removed? The `reason` in `rustc_deprecated` now seems to be ignored.

Comment (euclio) [reply src/test/ui/deprecation/atomic_initializers.stderr]: It hasn't been removed. The reason is already part of the error message.

Comment (varkor) [reply src/test/ui/deprecation/atomic_initializers.stderr]: Oh, I see: I didn't spot that a line was missing in GitHub's diff view.

File: src/librustc/middle/stability.rs [hunk 2]
```diff
@@ -576,7 +576,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                 if let hir::Node::Expr(_) = self.hir().get_by_hir_id(id) {
                     diag.span_suggestion(
                         span,
-                        &msg,
+                        "replace the usage of the deprecated item",
```
Comment (varkor) [inline src/librustc/middle/stability.rs]:
```suggestion
                        "replace the use of the deprecated item",
```

Comment (euclio) [reply src/librustc/middle/stability.rs]: Fixed.

**Pull request merged**